### ISSUE TITLE
Duplicate library search paths with --static --libs-only-L

### ISSUE THREAD

**Issue by avrahamshukron** [pkgconf/pkgconf#365]
Observed on version: 1.8.1 (Ubuntu 24.04)

Consider the following `gstreamer-1.0.pc`:
```pc
prefix=<some path>
exec_prefix=${prefix}
libdir=${prefix}/lib
includedir=${prefix}/include/gstreamer-1.0
toolsdir=${exec_prefix}/bin
pluginsdir=${prefix}/lib/gstreamer-1.0
datarootdir=${prefix}/share
datadir=${datarootdir}
girdir=${datadir}/gir-1.0
typelibdir=${libdir}/girepository-1.0

Name: GStreamer
Description: Streaming media framework
Version: 1.14.4
Requires: glib-2.0, gobject-2.0
Requires.private: gmodule-no-export-2.0  
Libs: -L${libdir} -lgstreamer-1.0
Cflags: -I${includedir}
```

Given the following command line:
```sh
pkg-config --libs-only-L --static gstreamer-1.0
```
Will produce the following output:
```
-L<prefix>/lib -L<prefix>/lib -L<prefix>/lib -L<prefix>/lib 
```
(4 paths because 1 for gstreamer and 3 for dependencies, and all the same because all are in the same libdir, which is typical).

It looks like the library search paths are not de-duped because of --static, even though it doesn't really make sense when only -L entries are requested.

In short - library search paths should always be de-duped.

**Comment by dg0yt** [comment]
> Observed on version: 1.8.1 (Ubuntu 24.04)

And does the problem also occur with 2.3.0? There was a lot of improvement since 1.8.1.

**Comment by avrahamshukron** [comment]
I have no way of testing on latest version ATM.
I'm bound to a specific build env, and 1.8.1 is what shipped in Ubuntu 24.04.

**Comment by kaniini** [comment]
pkgconf 1.x is end of life, if you can reproduce on pkgconf 2.x, reopen.
